feat: Added german language

Dacid99 opened the pull request:
Hi, 

I added the strings translations to german.
However I am unsure whether I put the strings.xml file in the right place (naming of the folder) and whether I did everything to add the option for german language to the language settings and for translation to work.
If you give me feedback I will try to implement the missing parts.

Thank you very much for this project, Im using it on my deggogled Samsung A40 and it works like a charm!

jd1378 commented on app/src/main/res/values-de/strings.xml:
```diff
@@ -0,0 +1,51 @@
+  <string name="failed_to_open_app_settings">Einstellungen öffnen fehlgeschlagen</string>
+  <string name="perm_restricted_desc">Falls du \"Benachrichtigungen lesen\" in Schritt 2 nicht erlauben konntest, versuche \"Eingeschränkte Einstellungen erlauben\" in den App Einstellungen</string>
+  <string name="en" translatable="false">English</string>
+  <string name="fa" translatable="false">فارسی</string>
```
add language code "de" entry under here

jd1378 replied:
(in general `translatable="false"` doesn't belong outside the main strings.xml but I will move it after merge, so np)

Dacid99 replied:
oh yup that makes sense, thanks for the feedback

jd1378 commented:
Hi
do you mind making a copy of files inside [fastlane/metadata/android/en-US](https://github.com/jd1378/otphelper/tree/main/fastlane/metadata/android/en-US) to `fastlane/metadata/android/de` and translate them (don't copy `images` folder) ?

Dacid99 commented:
Yeah sure no problem, I ll do it right when i get back home 👍

jd1378 commented:
Oh also before I forget:
- max character limit for app title is 30 character
- for short description it's 80

please make sure the translations for those two are within the limit
thank you

jd1378 commented on app/src/main/res/values-de/strings.xml:
```diff
@@ -0,0 +1,48 @@
+<resources>
+  <string name="app_name">Finde SMS Codes</string>
```
this is the app name shown on the phone launcher, please match it with the title.txt file in metadata folder (like "Copy SMS Code", if app title is going to stay English). this is to prevent confusion for new users who install the app.

Dacid99 replied:
sure, done

Dacid99 replied:
Funny, the app recognised the "txt" from the notification email i got for your comment as an OTP xD

Dacid99 commented:
Ok I think Im done.
I have 
- removed the `translatable="false"` parts in the values-de/strings.xml
- added the de label to the values/strings.xml
- copied the fastlane/metadata/android/en-US to fastlane/metadata/android/de
- translated all the files with regards to the char limit and the matching of app name

jd1378 commented:
thank you

jd1378 commented:
released in [v1.2.0](https://github.com/jd1378/otphelper/releases/tag/v1.2.0)

Dacid99 commented:
My pleasure, if there turn out to be an issue with this translation (e.g. text being to long for a box) just let me know, Im happy to fix it

Dacid99 commented:
ok ill check it out once i get the new version from fdroid

jd1378 commented:
Here is a screenshot of it:

![image](https://github.com/jd1378/otphelper/assets/6196533/9d24e7dc-bea3-4516-bb8c-0d70c695b790)

No need to change it though if the meaning is as should be
